libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -99,6 +103,30 @@ func (out *otelConsumer) logsPublish(ctx context.Context, batch publisher.Batch)
 
 	err := out.logsConsumer.ConsumeLogs(ctx, pLogs)
 	if err != nil {
+		// If the batch is too large, the elasticsearchexporter will
+		// return a 413 error.
+		//
+		// At the moment, the exporter does not support batch splitting
+		// on error so we do it here.
+		//
+		// See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36163.
+		if strings.Contains(err.Error(), "Request Entity Too Large") {
```

Comment:
I'm not 100% sure about this check:

1. How confident are we that this text would always be present in the error message?
2. What if we check for the presence of `413` instead?
3. Can you post here in the comment the sample of the full error message returned here?

Reply:
1. Since the error comes from ES, the error message should be the same every time in this case.
2. That is also a possibility, both will work in this case. This is a temporary solution and hopefully we can use a sentinel error for this case in the future.
3. The full error message is `flush failed (413): [413 Request Entity Too Large]`